niceman/retrace/packagemanagers.py

```diff
@@ -215,8 +215,16 @@ def _create_named_origin(o, used_names):
         # Remember the created name
         used_names.add(name)
         # Create a named origin
-        new_o = copy.deepcopy(o)
-        new_o.name = name
+        new_o = DpkgOrigin(name=name,
+                           component=o.component,
+                           codename=o.codename,
+                           archive=o.archive,
+                           architecture=o.architecture,
+                           origin=o.origin,
+                           label=o.label,
+                           site=o.site,
+                           archive_uri=o.archive_uri,
+                           date=o.date)
```

Comment:
there should be a better way ;)  e.g. if you go through all _attrs, compile a dict (e.g. params) with adjusted name and pass into `DpkgOrigin(**params)`